app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/weekly/WeeklyReceived.java

```diff
@@ -0,0 +1,65 @@
+package nyc.c4q.jonathancolon.inContaq.utlities.sms.weekly;
+
+import android.os.AsyncTask;
+import android.util.Log;
+import org.joda.time.DateTime;
+import java.util.ArrayList;
+import java.util.TreeMap;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+
+public class WeeklyReceived extends AsyncTask<WeeklyTask, Void, TreeMap<Integer, Integer>> {
+
+    private TreeMap<Integer, Integer> weeklyTexts;
+
+    public WeeklyReceived() {
+    }
+
+    @Override
+    protected void onPreExecute() {
+
+        Log.i("WeeklyReceivedTask", "Getting Weekly Total Received...");
+    }
+
+    @Override
+    protected TreeMap<Integer, Integer> doInBackground(WeeklyTask... params) {
+
+        ArrayList<Sms> listSms = params[0].listSms;
+        weeklyTexts = params[0].weeklyTexts;
```

Comment:
Why do you set weeklyTexts here? You overwrite it with your next call to getSmsStats()